packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -23,6 +24,63 @@ export interface IService extends IResource {
   readonly serviceArn: string;
 }
 
+/**
+ * Properties for ECS container targets.
+ */
+export interface TargetProps {
+  /**
+   * Name of the container.
+   */
+  readonly containerName: string,
+
+  /**
+   * Container targets of the container.
+   */
+  readonly containerTargets: ContainerTargetProps[]
+}
+
+/**
+ * Properties for registering a container target.
+ */
+export interface ContainerTargetProps {
+  /**
+   * Container port number to be registered.
+   */
+  readonly containerPort: number,
+
+  /**
+   * Protocol of the container port.
+   *
+   * @default Protocol.TCP
+   */
+  readonly protocol?: Protocol,
+
+  /**
+   * Setting to create and attach target groups.
+   */
+  readonly targetGroups: TargetGroupProps [],
+}
+
+/**
+ * The interface for creating and attaching a target group.
+ */
+export interface TargetGroupProps {
+  /**
+   * ID for a target group to be created.
+   */
+  readonly targetGroupId: string,
+
+  /**
+   * Properties for adding new targets to a listener.
+   */
+  readonly addTargetGroupProps?: elbv2.AddApplicationTargetsProps | elbv2.AddNetworkTargetsProps,
+
+  /**
+   * Listener for the target group to attach to [disable-awslint:ref-via-interface].
+   */
+  readonly listener: elbv2.ApplicationListener | elbv2.NetworkListener
```

Comment:
Why are the interfaces not good enough?

Can that be fixed?

Reply:
So it's because `INetworkListener` does not have the method `addTargets`, and arn is the only property the interface has.

Reply:
Sure. Would it be possible to effectively implement `addTargets()` in the interfaces?

I guess not, huh?

Reply:
It is still possible to implement addTargets() but it will be a little bit clumsy and reinvent the wheels. Yeah I think you are right.